### PR TITLE
Resolve or supress all 800-53r4 checks in ATAT stack

### PR DESCRIPTION
**Pull request by kylelaker-ccpo**
We still need to do some work to solve IAM and Network; however, this
fixes the issues in the main app stack. All remaining issues are
resolved or are suppressed and the check is added to the deployment
pipeline to ensure that we always remain compliant. This is far easier
to do now that we have networking solved (except that tricky
`AwsCustomResource`).

This also adds a synthesis of a full environment to our CI workflow.

### CODE REVIEW

File: lib/atat-web-api-stack.ts [hunk 1]
```diff
@@ -59,14 +61,19 @@ export class AtatWebApiStack extends cdk.Stack {
 
   constructor(scope: Construct, id: string, props: AtatWebApiStackProps) {
     super(scope, id, props);
+
     const { environmentName, network } = props;
+    const sharedAtatKmsKey = new kms.Key(this, "AtatKey", {
+      enableKeyRotation: true,
+    });
```
Comment (juliusfitzhugh-ccpo) [inline lib/atat-web-api-stack.ts:68]: I know we mentioned rotating keys using lambda, is this the same or different from that?

File: lib/atat-web-api-stack.ts [hunk 2]
```diff
@@ -81,7 +88,16 @@ export class AtatWebApiStack extends cdk.Stack {
       const loadBalancer = new elbv2.ApplicationLoadBalancer(this, "LoadBalancer", {
         vpc: network.vpc,
         internetFacing: false,
+        deletionProtection: true,
```
Comment (juliusfitzhugh-ccpo) [inline lib/atat-web-api-stack.ts:91]: Nice add!

**Review by juliusfitzhugh-ccpo — APPROVED** [review]
Looks good! Thanks for providing reasons for the suppressions, it helps give context of other things that are in play outside of AWS.

File: lib/atat-web-api-stack.ts (hunk 1 above)
Comment (kylelaker-ccpo) [reply lib/atat-web-api-stack.ts:68]: This is different -- I could try to explain it but honestly I think the graphics at https://docs.aws.amazon.com/kms/latest/developerguide/rotate-keys.html explain it a little better than I can. This is purely for the KMS Key, not for any sort of larger definition of "key"

File: lib/atat-web-api-stack.ts (hunk 2 above)
Comment (kylelaker-ccpo) [reply lib/atat-web-api-stack.ts:91]: Yeah, it'll be nice to make sure we don't change the DNS name unexpectedly. This is actually one of the `cdk-nag` rules.

**Review by jeffsegal-ccpo — APPROVED** [review]

**Pull request merged**